edgelet/dps/src/registration.rs

```diff
@@ -442,52 +442,54 @@ fn get_device_info(
 mod tests {
     use super::*;
 
-    use std::cell::RefCell;
-    use std::mem;
+    use std::{mem, sync::Mutex};
 
-    use hyper::header::Authorization;
-    use hyper::server::service_fn;
-    use hyper::StatusCode;
+    use http;
+    use hyper::{self, Body, Request, Response, StatusCode};
     use serde_json;
-    use tokio_core::reactor::Core;
+    use tokio;
     use url::Url;
 
     use edgelet_core::crypto::{MemoryKey, MemoryKeyStore};
 
     #[test]
     fn server_register_with_auth_success() {
-        let mut core = Core::new().unwrap();
         let expected_uri = "https://global.azure-devices-provisioning.net/scope/registrations/reg/register?api-version=2017-11-15";
-        let handler = move |req: Request| {
-            let (method, uri, _httpversion, headers, _body) = req.deconstruct();
+        let handler = move |req: Request<Body>| {
+            let (
+                http::request::Parts {
+                    method,
+                    uri,
+                    headers,
+                    ..
+                },
+                _body,
+            ) = req.into_parts();
             assert_eq!(uri, expected_uri);
-            assert_eq!(method, Method::Put);
+            assert_eq!(method, Method::PUT);
             // If authorization header does not have the shared access signature, request one
-            let auth = headers.get::<Authorization<String>>();
+            let auth = headers.get(hyper::header::AUTHORIZATION);
             match auth {
                 None => {
                     let mut result = TpmRegistrationResult::new();
                     result.set_authentication_key(base64::encode("key"));
-                    future::ok(
-                        Response::new()
-                            .with_status(StatusCode::Unauthorized)
-                            .with_body(serde_json::to_string(&result).unwrap().into_bytes()),
-                    )
+                    let mut response =
+                        Response::new(serde_json::to_string(&result).unwrap().into());
+                    *response.status_mut() = StatusCode::UNAUTHORIZED;
```

Comment:
I think I did this because the Response builder enforces having to deal with a Result - `Builder::build()` returns a Result in case you configured it wrong. Will double-check.

Reply:
It's a test using `unwrap()` anyway, so I changed it to use the builder.